File: frontend/src/constants/provincias.ts
export const provinciasOptions = [
  { value: "Buenos Aires", label: "Buenos Aires" },
  { value: "CABA", label: "Ciudad Autónoma de Buenos Aires" },
  { value: "Catamarca", label: "Catamarca" },
  { value: "Chaco", label: "Chaco" },
  { value: "Chubut", label: "Chubut" },
  { value: "Córdoba", label: "Córdoba" },
  { value: "Corrientes", label: "Corrientes" },
  { value: "Entre Ríos", label: "Entre Ríos" },
  { value: "Formosa", label: "Formosa" },
  { value: "Jujuy", label: "Jujuy" },
  { value: "La Pampa", label: "La Pampa" },
  { value: "La Rioja", label: "La Rioja" },
  { value: "Mendoza", label: "Mendoza" },
  { value: "Misiones", label: "Misiones" },
  { value: "Neuquén", label: "Neuquén" },
  { value: "Río Negro", label: "Río Negro" },
  { value: "Salta", label: "Salta" },
  { value: "San Juan", label: "San Juan" },
  { value: "San Luis", label: "San Luis" },
  { value: "Santa Cruz", label: "Santa Cruz" },
  { value: "Santa Fe", label: "Santa Fe" },
  { value: "Santiago del Estero", label: "Santiago del Estero" },
  { value: "Tierra del Fuego", label: "Tierra del Fuego" },
  { value: "Tucumán", label: "Tucumán" },
];

const provinciasValues = new Set(provinciasOptions.map((option) => option.value));
const provinciasPorClave = new Map(
  provinciasOptions.flatMap((option) => {
    const aliases =
      option.value === "CABA"
        ? ["CABA", "CIUDAD AUTONOMA DE BUENOS AIRES", "CAPITAL FEDERAL"]
        : [option.value];
    return aliases.map((alias) => [normalizarClave(alias), option.value]);
  }),
);

export const esProvinciaArgentina = (value: string | null): value is string =>
  !!value && (provinciasValues.has(value) || provinciasPorClave.has(normalizarClave(value)));

export const normalizarProvinciaArgentina = (
  value: string | null,
): string | null => {
  if (!value) return null;
  return provinciasValues.has(value)
    ? value
    : provinciasPorClave.get(normalizarClave(value)) || null;
};

function normalizarClave(value: string): string {
  return value
    .normalize("NFD")
    .replace(/[\u0300-\u036f]/g, "")
    .replace(/\s+/g, " ")
    .trim()
    .toUpperCase();
}

File: backend/app/core/provincias.py
"""Catalogo de provincias argentinas."""

from __future__ import annotations

import re
import unicodedata


PROVINCIAS_ARGENTINAS = {
    "BUENOS AIRES": "Buenos Aires",
    "CABA": "CABA",
    "CIUDAD AUTONOMA DE BUENOS AIRES": "CABA",
    "CIUDAD AUTÓNOMA DE BUENOS AIRES": "CABA",
    "CAPITAL FEDERAL": "CABA",
    "CATAMARCA": "Catamarca",
    "CHACO": "Chaco",
    "CHUBUT": "Chubut",
    "CORDOBA": "Córdoba",
    "CÓRDOBA": "Córdoba",
    "CORRIENTES": "Corrientes",
    "ENTRE RIOS": "Entre Ríos",
    "ENTRE RÍOS": "Entre Ríos",
    "FORMOSA": "Formosa",
    "JUJUY": "Jujuy",
    "LA PAMPA": "La Pampa",
    "LA RIOJA": "La Rioja",
    "MENDOZA": "Mendoza",
    "MISIONES": "Misiones",
    "NEUQUEN": "Neuquén",
    "NEUQUÉN": "Neuquén",
    "RIO NEGRO": "Río Negro",
    "RÍO NEGRO": "Río Negro",
    "SALTA": "Salta",
    "SAN JUAN": "San Juan",
    "SAN LUIS": "San Luis",
    "SANTA CRUZ": "Santa Cruz",
    "SANTA FE": "Santa Fe",
    "SANTIAGO DEL ESTERO": "Santiago del Estero",
    "TIERRA DEL FUEGO": "Tierra del Fuego",
    "TUCUMAN": "Tucumán",
    "TUCUMÁN": "Tucumán",
}


def normalizar_provincia_argentina(value: str | None) -> str | None:
    """Normalizar una provincia argentina a su etiqueta canonica."""
    if not value:
        return None

    normalizada = _normalizar_clave(value)
    return PROVINCIAS_ARGENTINAS.get(normalizada)


def _normalizar_clave(value: str) -> str:
    """Normalizar texto para busqueda tolerante de provincias."""
    sin_acentos = unicodedata.normalize("NFKD", value)
    sin_marcas = "".join(
        char for char in sin_acentos if not unicodedata.combining(char)
    )
    return re.sub(r"\s+", " ", sin_marcas.upper()).strip()
